CodeAnalysisTreatWarningsAsErrors does not work in Visual Studio

**Version Used**: 
- **VS:** 16.8.0 Preview 4.0 [30518.80.main]
- **.NET SDK:** 5.0.100-rc.2.20468.8

**Steps to Reproduce**:

Create a project with .NET 5 RC2 with the following project file and code file:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <CodeAnalysisTreatWarningsAsErrors>false</CodeAnalysisTreatWarningsAsErrors>
    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  </PropertyGroup>

</Project>
```

```csharp
using System;

class C
{
    static void Main(string[] str)
    {
        int length = 3;
        for (int i = 0; i < length; i++)
        {
            Span<int> numbers = stackalloc int[length]; // CA2014
            numbers[i] = i;
        }
    }
}

```

**Expected Behavior**:

CA2014 is treated as a warning everywhere.

**Actual Behavior**:

It is treated as an error _only_ in visual studio
![image](https://user-images.githubusercontent.com/9797472/93640593-b3d69900-f9af-11ea-9839-e83a11f4a844.png)

the commandline treats it as a warning:
```cmd
dotnet build
Microsoft (R) Build Engine version 16.8.0-preview-20464-02+0220c5eae for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  All projects are up-to-date for restore.
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
C:\source\jmarolf\scratch\analyzer-test\Program.cs(10,33): warning CA2014: Potential stack overflow. Move the stackalloc out of the loop. [C:\source\jmarolf\scratch\analyzer-test\analyzer-test.csproj]
  analyzer-test -> C:\source\jmarolf\scratch\analyzer-test\bin\Debug\net5.0\analyzer-test.dll

Build succeeded.

C:\source\jmarolf\scratch\analyzer-test\Program.cs(10,33): warning CA2014: Potential stack overflow. Move the stackalloc out of the loop. [C:\source\jmarolf\scratch\analyzer-test\analyzer-test.csproj]
    1 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.98
```

--- a/src/Tools/GenerateDocumentationAndConfigFiles/Program.cs
+++ b/src/Tools/GenerateDocumentationAndConfigFiles/Program.cs
@@ -262,8 +262,9 @@ $@"<Project>
     This property group prevents the rule ids implemented in this package to be bumped to errors when
     the 'CodeAnalysisTreatWarningsAsErrors' = 'false'.
   -->
-  <PropertyGroup Condition=""'$(CodeAnalysisTreatWarningsAsErrors)' == 'false'"">
-    <WarningsNotAsErrors>$(WarningsNotAsErrors);{allRuleIds}</WarningsNotAsErrors>
+  <PropertyGroup>
+    <CodeAnalysisRuleIds>{allRuleIds}</CodeAnalysisRuleIds>
+    <WarningsNotAsErrors Condition=""'$(CodeAnalysisTreatWarningsAsErrors)' == 'false'"">$(WarningsNotAsErrors);$(CodeAnalysisRuleIds)</WarningsNotAsErrors>
   </PropertyGroup>";
             }
 

--- a/src/Tools/GenerateGlobalAnalyzerConfigs/Program.cs
+++ b/src/Tools/GenerateGlobalAnalyzerConfigs/Program.cs
@@ -371,7 +371,7 @@ $@"<Project>{GetCommonContents(packageName)}{GetPackageSpecificContents(packageN
             File.WriteAllText(fileWithPath, fileContents);
 
             static string GetCommonContents(string packageName)
-                => GetGlobalAnalyzerConfigTargetContents(packageName) + GetMSBuildContentForPropertyAndItemOptions();
+                => GetGlobalAnalyzerConfigTargetContents(packageName) + GetMSBuildContentForPropertyAndItemOptions() + GetCodeAnalysisTreatWarningsAsErrorsTargetContents();
 
             static string GetGlobalAnalyzerConfigTargetContents(string packageName)
             {
@@ -475,6 +475,22 @@ $@"<Project>{GetCommonContents(packageName)}{GetPackageSpecificContents(packageN
 
                     AddItemGroupForCompilerVisibleProperties(compilerVisibleProperties, builder);
                 }
+            }
+
+            static string GetCodeAnalysisTreatWarningsAsErrorsTargetContents()
+            {
+                return $@"
+  <!--
+    Design-time target to prevent the rule ids implemented in this package to be bumped to errors in the IDE
+    when 'CodeAnalysisTreatWarningsAsErrors' = 'false'. Note that a similar 'WarningsNotAsErrors'
+    property group is present in the generated props file to ensure this functionality on command line builds.
+  -->
+  <Target Name=""_CodeAnalysisTreatWarningsNotAsErrors"" BeforeTargets=""CoreCompile"" Condition=""'$(CodeAnalysisTreatWarningsAsErrors)' == 'false' AND ('$(DesignTimeBuild)' == 'true' OR '$(BuildingProject)' != 'true')"">
+    <PropertyGroup>
+      <WarningsNotAsErrors>$(WarningsNotAsErrors);$(CodeAnalysisRuleIds)</WarningsNotAsErrors>
+    </PropertyGroup>    
+  </Target>
+";
             }
 
             static string GetPackageSpecificContents(string packageName)

